### PR TITLE
Wiki collaborators feature

### PR DESCRIPTION
**Pull request by ARodriguezHacks**
Requesting to merge wiki-collaborators branch into master branch of Blocipedia. I first needed to figure out how to merge my wiki-collaborators branch on Github with my local wiki-collaborators branch. I made fixes to a few merge conflicts and then made a new git commit before pushing the merged branch back upstream.

### CODE REVIEW

File: app/controllers/collaborators_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class CollaboratorsController < ApplicationController
+
+  def create
+    @wiki = Wiki.find(params[:wiki_id])
+    @user = User.find_by(email: params[:collaborator][:user])
+
+    if @user
```
Comment (alexstophel) [inline app/controllers/collaborators_controller.rb]: Consider using a method like `.present?` on the `@user` variable here. If possible, we'd like conditionals to make decisions based on true/false values.

File: app/policies/wiki_policy.rb [hunk 2]
```diff
@@ -1,9 +1,33 @@
...
+    def initialize(user,scope)
+      @user = user
+      @scope = scope
+    end
+
     def resolve
-      if user.present?
-        scope.all
+      wikis = []
+      if user.role == 'admin'
```
Comment (alexstophel) [inline app/policies/wiki_policy.rb]: You can just call `user.admin?` I believe as well.

File: app/policies/wiki_policy.rb [hunk 3]
```diff
@@ -1,9 +1,33 @@
...
+        wikis = scope.all
+      elsif user.role == 'premium'
```
Comment (alexstophel) [inline app/policies/wiki_policy.rb]: `user.premium?`

File: app/views/collaborators/_delete.html.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<h4>Collaborators</h4>
+<ul>
+  <% @wiki.collaborators.each do |c| %>
+    <li>
+      <%= User.find_by_id(c.user_id).email %>
```
Comment (alexstophel) [inline app/views/collaborators/_delete.html.erb]: Instead of doing a lookup here, does `c.user.email` work?

Comment (ARodriguezHacks) [reply app/views/collaborators/_delete.html.erb]: I just tested it, and yes it does work.

**Pull request merged**